### PR TITLE
Fixed issue when removing handlers on filtered events

### LINKED ISSUE #2684: Failed to remove handler attached on filtered event
## 🐛 Bug description

```python
from ignite.engine import Engine, Events

e = Engine(lambda e, b: None)
assert len(e._event_handlers[Events.ITERATION_COMPLETED]) == 0

handle = e.add_event_handler(Events.ITERATION_COMPLETED(every=3), lambda _: print("Hello"))
assert len(e._event_handlers[Events.ITERATION_COMPLETED]) == 1

handle.remove()
assert len(e._event_handlers[Events.ITERATION_COMPLETED]) == 0, e._event_handlers[Events.ITERATION_COMPLETED]
```
Output:
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Input In [8], in <cell line: 11>()
      8 assert len(e._event_handlers[Events.ITERATION_COMPLETED]) == 1
     10 handle.remove()
---> 11 assert len(e._event_handlers[Events.ITERATION_COMPLETED]) == 0, e._event_handlers[Events.ITERATION_COMPLETED]

AssertionError: [(<function <lambda> at 0x132ab23a0>, (<ignite.engine.engine.Engine object at 0x13330ed00>,), {})]
> 
```

## Environment

- PyTorch Version (e.g., 1.4):
- Ignite Version (e.g., 0.3.0): 0.4.9
- OS (e.g., Linux):
- How you installed Ignite (`conda`, `pip`, source):
- Python version:
- Any other relevant information:

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixes #2684

Description:
- Fixed issue when removing handlers on filtered events
- Updated tests
- Some code formatting to reduce LOC

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/engine/test_event_handlers.py [hunk 1]
```diff
@@ -129,73 +129,86 @@ def test_adding_multiple_event_handlers():
...
 
     removable_handle.remove()
-    assert not engine.has_event_handler(handler, Events.STARTED)
+    assert not engine.has_event_handler(handler, event1)
 
     # Second engine pass does not fire handle again.
-    engine.run(1)
-    handler.assert_called_once_with(engine)
+    engine.run([1, 2])
+    handler.assert_any_call(engine)
```
Comment (sadra-barikbin) [inline tests/ignite/engine/test_event_handlers.py:159]: This check does not have a benefit this way. Instead we can:
```python
    assert handler.call_count == call_count_in_line_152
```

File: tests/ignite/engine/test_event_handlers.py [hunk 2]
```diff
@@ -129,73 +129,86 @@ def test_adding_multiple_event_handlers():
...
 
-    assert not engine.has_event_handler(handler, Events.STARTED)
-    handler.assert_called_once_with(engine)
+    assert not engine.has_event_handler(handler, event1)
+    handler.assert_any_call(engine)
 
-    engine.run(1)
-    handler.assert_called_once_with(engine)
+    engine.run([1, 2])
+    handler.assert_any_call(engine)
```
Comment (sadra-barikbin) [inline tests/ignite/engine/test_event_handlers.py:172]: Here as well

File: tests/ignite/engine/test_event_handlers.py (hunk 1 above)
Comment (vfdev-5) [reply tests/ignite/engine/test_event_handlers.py:159]: This check is optional as the test about removing the handler. `call_count_in_line_152` is not tricky to get, we could put it into parametrization, but I think we can keep it as it is as the test is mainly about another thing.

Comment (sadra-barikbin) [reply tests/ignite/engine/test_event_handlers.py:159]: I mean this test anyway passes even if removing handler does not work. Is it OK?

Comment (vfdev-5) [reply tests/ignite/engine/test_event_handlers.py:159]: I see what you mean, yes you are right. I'll update that.

**Review by sadra-barikbin — APPROVED** [review]

**Pull request merged**